nexuse2e-core/src/main/java/org/nexuse2e/ui2/rest/TransactionReportingHandler.java

```diff
@@ -479,4 +490,62 @@ private void returnEngineTimeVariables(HttpServletResponse response) throws JSON
 
         response.getOutputStream().print(String.valueOf(variables));
     }
+
+    private boolean neverOrTooLongAgo(Date date) {
+        return date == null || date.before(getCurrentDateMinusWeeks(transactionActivityTimeframeInWeeks));
+    }
+
+    private String formatTimeDifference(Date date) {
+        if (neverOrTooLongAgo(date)) {
+            return "no messages in " + transactionActivityTimeframeInWeeks + " weeks";
```

Comment:
Is generating the message something that the server should do or is it better to do it on the client and use something like moment.js to get this relative time format?

Reply:
These values are only numbers and not a date format, building this return message here is okay.